### PR TITLE
i18n Banner

### PR DESCRIPTION
**Pull request by mateoclarke**
_Addresses part of #21. Punts on most of #35_

This isn't totally functional yet. Just mainly focusing on the markup. The language choice doesn't persist and it is only stored in the state of this component when it should probably be managed at a higher level.

We still need to: 
- update the API call to pull down language appropriate data onto the page
- and/or we need to reroute the user to a url with with the language code appended, ex: `austin.gov/es`
- The dropdown isn't a great UX, but I wanted to talk to Celine about what she expects from it before spending more time on it. _edit: now remembering about #35 where its better defined_

Not sure about the best long term approach there so I'm going to punt until we can investigate if Next.js has some patterns we should look into.

screen shot 2017-12-20 at 6 07 48 pm

screen shot 2017-12-20 at 6 08 13 pm

### CODE REVIEW

File: src/components/I18nBanner.js [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    title: 'မြန်မာ',
+    abbr: 'my',
+  },
+]
+
+class I18nBanner extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeLanguage: locale().substring(0,2) || 'en', // we could
```
Comment (benweatherman) [inline src/components/I18nBanner.js]: could we?

File: src/components/I18nBanner.js [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+  setLangauage = (e) => {
+    e.stopPropagation();
+    e.nativeEvent.stopImmediatePropagation();
+    const languageAbbr = e.target.parentElement.lang;
+    this.setState({
+      activeLanguage: languageAbbr,
+      isOpen: false,
+    })
+    console.log(this.state.activeLanguage)
```
Comment (benweatherman) [inline src/components/I18nBanner.js]: Definitely not something for right now: if/when we do server redirects for language we should set a cookie here so the server will know the user has overridden the language from the browser settings.

File: src/components/I18nBanner.js (hunk 1 above)
Comment (mateoclarke) [reply src/components/I18nBanner.js]: image

File: src/components/I18nBanner.js [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+  otherLanguageClassName = () => {
+    const base = `coa-I18nBanner__other-language-options`;
+    return this.state.isOpen ? `${base}--open` : `${base}--closed`;
+  }
+
+  setLangauage = (e) => {
+    e.stopPropagation();
+    e.nativeEvent.stopImmediatePropagation();
+    const languageAbbr = e.target.parentElement.lang;
```
Comment (ifsimicoded) [inline src/components/I18nBanner.js]: this way of getting the element that's clicked seems highly coupled to the dom structure. maybe try e.currentTarget to get the element which had the handler.

File: src/components/I18nBanner.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    abbr: 'en',
+  }, {
+    title: 'Español',
+    abbr: 'es',
+  }, {
+    title: 'Tiếng Việt',
+    abbr: 'vn',
+  }, {
+    title: '中文',
+    abbr: 'zh',
```
[the remark this answers is not in the document]
Comment (mateoclarke) [reply src/components/I18nBanner.js]: Yeah, this is a good point. I didn't get all the way to the end of my research about the form of Chinese we'll be using.

This is the best list of language codes I've found: http://www.iana.org/assignments/language-subtag-registry/language-subtag-registry

And its seems like Simplified and Traditional Chinese share the `zh` language code, but have subtags to distinguish them. Not sure what that means for us in practice, if we're really going to translate for both, or how we would deal with language codes that aren't just two characters. 

Thus, I just kept their abbreviation the same.

[the remark this answers is not in the document]
Comment (mateoclarke) [reply src/components/I18nBanner.js]: Created an issue: https://github.com/cityofaustin/janis/issues/53 For now, I think I'll just comment out Traditional Chinese for simplicity.

File: src/css/components/I18nBanner.scss [hunk 5]
```diff
@@ -0,0 +1,74 @@
+.coa-I18nBanner {
+  width: 100%;
+  background-color: $color-gray-light;
+  padding: .7rem 0;
```
[the remark this answers is not in the document]
Comment (mateoclarke) [reply src/css/components/I18nBanner.scss]: ✅

File: src/css/components/I18nBanner.scss [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+.coa-I18nBanner__language-list {
+  text-align: right;
+}
+
+.coa-I18nBanner__prompt {
+  display: none;
```
[the remark this answers is not in the document]
Comment (mateoclarke) [reply src/css/components/I18nBanner.scss]: ✅

File: src/css/components/I18nBanner.scss [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  font-size: 1.5rem;
```
[the remark this answers is not in the document]
Comment (mateoclarke) [reply src/css/components/I18nBanner.scss]: cool. Do we want to define these in `_variables_coa.scss`? For now, I'm using the font size values set in `_variables_uswds.scss`

File: src/svg/CaretDown.js [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+
+function CaretDown(props) {
+  const { size } = props
+
+  return (
+    <svg width={size} height={size} viewBox="0 0 1792 1792" xmlns="http://www.w3.org/2000/svg" aria-labelledby="title">
```
[the remark this answers is not in the document]
Comment (mateoclarke) [reply src/svg/CaretDown.js]: remind me when we are both back in the office. no magic here.

**Pull request merged**